datadreamer/prompt_generation/lm_prompt_generator.py

```diff
@@ -94,6 +119,20 @@ def _create_lm_prompt_text(self, selected_objects: List[str]) -> str:
         """
         return f"[INST] Generate a short and concise caption for an image. Follow this template: 'A photo of {', '.join(selected_objects)}', where the objects interact in a meaningful way within a scene, complete with a short scene description. [/INST]"
 
+    def _test_prompt(self, prompt: str, selected_objects: List[str]) -> bool:
+        """Tests if the generated prompt is valid based on selected objects.
+
+        Args:
+            prompt (str): The generated prompt.
+            selected_objects (List[str]): Objects to check in the prompt.
+
+        Returns:
+            bool: True if the prompt is valid, False otherwise.
+        """
+        return prompt.lower().startswith(
+            "a photo of"
+        )  # and all(obj.lower() in prompt.lower() for obj in selected_objects)
```

Comment:
@HonzaCuhel do you have any other ideas on how to ensure the prompt follows the template?

Reply:
Well, when I was experimenting with TinyLlama I tried several approaches. I tried to command the LM to generate the response that follows this template through the prompt. However, the generated responses weren't satisfactory enough. I also tried to edit the prompt in such a way that I explicitly wrote the words "A photo of" after the instruction end token so that the generated responses started with the template; however, in the case of TinyLlama (I haven't tried it with Mistral) it confused the LM, and it was generating gibberish. There are other ways to restrict the generated tokens; however, when I tried them some time ago, they weren't working very well, but that could change. 

When I was experimenting with TinyLlama, I noticed that sometimes the generated prompts were still usable even though they didn't pass the test, e.g. ("A picture of ..." or "Picture showing ..." etc.). Of course, sometimes the prompts weren't good, so this test ensures at least some quality. So maybe we could also include in this test these cases "A picture of," etc., but maybe first try to measure how often this happens and if it is worth adding it. 

I could look more into it, but I generally think we should have some simple regex tests to ensure the quality of the generated prompts for their simplicity, the fact that they are fast, and also that they are relatively powerful.